appservice/src/utils/zip.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import { Readable } from 'stream';
+import * as yazl from 'yazl';
+
+export async function zip(deployPath: string, files: string[]): Promise<Readable> {
+    const zipFile = new yazl.ZipFile();
+    const buffer: string[] = [];
+    zipFile.outputStream.on('data', (data) => {
```

Comment:
It honestly looks like you can return `zipFile.outputStream` (If you change the return type to `NodeJS.ReadableStream`) and you have to do `.on` at all. It's already a stream.

Reply:
Also once you get rid of the `.on` stuff, this is a pretty simple function and I guess I'd rather have the code in `runWithZipStream` instead of a utils file

Reply:
`NodeJS.ReadableStream` wasn't actually compatible with `Stream` which is the type that we are using so I didn't want to carelessly change the types.

I originally did just pass `zip.outputStream` but I don't think it was working (though that may have been related to a different issue).

Reply:
It breaks this specifically https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/deploy/deployToStorageAccount.ts#L82

Reply:
Okay seems like we want this

```
return new Readable().wrap(zipFile.outputStream);
```


https://nodejs.org/api/stream.html#stream_readable_wrap_stream